### DISCUSSION TITLE
App Architecture

### DISCUSSION THREAD

**Discussion by mboyamike** [droidconKE/flutterconKEApp#10] [Ideas]
How should we approach the app architecture? 

Here's my proposal:

**App Architecture**
I propose having 3 layers for the app. The UI layer having all the widgets and screens. The logic layer which will have the state management item of choice (Blocs, Notifiers, providers etc.) and a data layer that will have repositories. 

I actively propose avoiding clean architecture

**Folder Structure**
Layer then feature.
This I think would be easy to communicate to contributors. So we'll have something like:

lib
  -  providers
      -  authentication
      - home
  - ui
      - authentication
      - home
      - widgets // shared app wide

**Comment by ChristineWasike** [comment]
This sounds simple enough. I have no issue with this

**Comment by MillerAdulu** [comment]
We will need to make an issue from this once #90 closes to really clean up the architecture and folder structure.
